cpu/x86/x86_pci.c

```diff
@@ -181,7 +181,7 @@ static void pci_setup_ios(struct x86_known_pci_device *dev)
                 printf("    BAR %u: memory, physical = 0x%08x-0x%08x, virtual = 0x%08x-0x%08lx\n",
                        bar_num,
                        physical_start, physical_start + length - 1,
-                       (unsigned) ptr, (uintptr_t) ptr + length - 1);
+                       (unsigned) ptr, (unsigned long) ((uintptr_t) ptr + length - 1));
```

Comment:
hm, not three pointers get printed by ```%x```, one by ```%lx```. If ```unsigned``` is fine for the other pointers, I suggest using ```%x``` for all and change the cast to just ```(unsigned)```